domain/src/main/java/com/teamagam/gimelgimel/domain/layers/ProcessVectorLayersInteractor.java

```diff
@@ -0,0 +1,127 @@
+import com.teamagam.gimelgimel.domain.base.rx.RetryWithDelay;
+import com.teamagam.gimelgimel.domain.config.Constants;
+import com.teamagam.gimelgimel.domain.layers.entitiy.VectorLayer;
+import com.teamagam.gimelgimel.domain.layers.entitiy.VectorLayerVisibilityChange;
+import com.teamagam.gimelgimel.domain.layers.repository.VectorLayersRepository;
+import com.teamagam.gimelgimel.domain.layers.repository.VectorLayersVisibilityRepository;
+import com.teamagam.gimelgimel.domain.messages.entity.ChatMessage;
+import com.teamagam.gimelgimel.domain.messages.entity.features.AlertFeature;
+import com.teamagam.gimelgimel.domain.messages.repository.MessagesRepository;
+import io.reactivex.Observable;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Date;
+import java.util.UUID;
+import javax.inject.Inject;
+
+public class ProcessVectorLayersInteractor extends BaseDataInteractor {
+
+  private static final Logger sLogger =
+      LoggerFactory.create(ProcessVectorLayersInteractor.class.getSimpleName());
+  private static final int MINIMUM_OFFSET = 1;
+  private static final String EMPTY_STRING = "";
+  private static final int VECTOR_LAYER_ALERT_SEVERITY = 1;
+  private static final String VECTOR_LAYER_ALERT_SOURCE = "SELF_GENERATED";
+  private final LayersLocalCache mLayersLocalCache;
+  private final VectorLayersRepository mVectorLayersRepository;
+  private final VectorLayersVisibilityRepository mVectorLayersVisibilityRepository;
+  private final MessagesRepository mMessagesRepository;
+  private final AlertsRepository mAlertsRepository;
+
+  @Inject
+  ProcessVectorLayersInteractor(ThreadExecutor threadExecutor,
+      LayersLocalCache layersLocalCache,
+      VectorLayersRepository vectorLayerRepository,
+      VectorLayersVisibilityRepository vectorLayersVisibilityRepository,
+      MessagesRepository messagesRepository, AlertsRepository alertsRepository) {
+    super(threadExecutor);
+    mLayersLocalCache = layersLocalCache;
+    mVectorLayersRepository = vectorLayerRepository;
+    mVectorLayersVisibilityRepository = vectorLayersVisibilityRepository;
+    mMessagesRepository = messagesRepository;
+    mAlertsRepository = alertsRepository;
+  }
+
+  @Override
+  protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+    DataSubscriptionRequest dataSubscriptionRequest =
+        factory.create(mVectorLayersRepository.getVectorLayersObservable(),
+            vlObservable -> vlObservable.map(
+                this::cacheLayer) // mapping in order to wait for the caching procedure
+                .doOnNext(this::setVisible)
+                .doOnNext(this::alertIfNeeded)
+                .retryWhen(new RetryWithDelay(Constants.LAYER_CACHING_RETRIES,
+                    Constants.LAYER_CACHING_RETRIES_DELAY_MS))
+                .doOnError(this::logFailure)
+                .onErrorResumeNext(Observable.empty()));
```

Comment:
If you'd have known how much trouble you caused by this little cute observation... Blimey!